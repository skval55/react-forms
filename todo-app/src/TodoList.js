import React, { useState } from "react";
import NewTodoForm from "./NewTodoForm";
import Todo from "./Todo";
import { v4 as uuid } from "uuid";

const TodoList = () => {
  const [list, setList] = useState([]);

  const renderTodoList = () => {
    return (
      <ul>
        {list.map((task) => (
          <Todo
            key={task.id}
            id={task.id}
            todo={task.todo}
            handleDelete={handleDelete}
          />
        ))}
      </ul>
    );
  };

  const addTask = (task) => {
    let newTask = { ...task, id: uuid() };
    console.log(newTask.id);
    setList((list) => [...list, newTask]);
  };

  function handleDelete(id) {
    setList((list) => list.filter((task) => task.id !== id));
  }

  return (
    <>
      <h1>To do list</h1>
      <NewTodoForm addTask={addTask} />
      {renderTodoList()}
    </>
  );
};

export default TodoList;
