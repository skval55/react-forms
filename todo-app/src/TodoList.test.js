import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

// smoke test
test("renders without crashing", () => {
  render(<TodoList />);
});

// snapshot test
it("matches snapshot", () => {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("can add a new item", function () {
  const { getByLabelText, queryByText } = render(<TodoList />);

  // no items yet
  expect(queryByText("feed cats")).not.toBeInTheDocument();

  const todoInput = getByLabelText("Todo:");
  const submitBtn = queryByText("Add");

  // fill out the form
  fireEvent.change(todoInput, { target: { value: "feed cats" } });
  fireEvent.click(submitBtn);

  // item exists!
  expect(queryByText("feed cats")).toBeInTheDocument();
});
it("can delete an item", function () {
  const { getByLabelText, queryByText } = render(<TodoList />);

  // no items yet
  expect(queryByText("feed cats")).not.toBeInTheDocument();

  const todoInput = getByLabelText("Todo:");
  const submitBtn = queryByText("Add");

  // fill out the form
  fireEvent.change(todoInput, { target: { value: "feed cats" } });
  fireEvent.click(submitBtn);

  // item exists!
  expect(queryByText("feed cats")).toBeInTheDocument();

  //   delete item
  const delButton = queryByText("X");
  fireEvent.click(delButton);

  //   deleted!
  expect(queryByText("feed cats")).not.toBeInTheDocument();
});
