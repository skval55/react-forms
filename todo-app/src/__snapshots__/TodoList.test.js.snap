// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <h1>
    To do list
  </h1>
  <form>
    <label
      for="todo"
    >
      Todo:
    </label>
    <input
      id="todo"
      name="todo"
      value=""
    />
    <button>
      Add
    </button>
  </form>
  <ul />
</DocumentFragment>
`;
